MediaBrowser.Model/Cryptography/PasswordHash.cs

```diff
@@ -0,0 +1,110 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace MediaBrowser.Model.Cryptography
+{
+    public class PasswordHash
+    {
+        //Defined from this hash storage spec
+        //https://github.com/P-H-C/phc-string-format/blob/master/phc-sf-spec.md
+        //$<id>[$<param>=<value>(,<param>=<value>)*][$<salt>[$<hash>]]
+
+        public string Id;
+        public Dictionary<string, string> Parameters = new Dictionary<string, string>();
+        public string Salt;
+        public byte[] SaltBytes;
+        public string Hash;
+        public byte[] HashBytes;
+        public PasswordHash(string StorageString)
+        {
+            string[] a = StorageString.Split('$');
+            Id = a[1];
+            if (a[2].Contains("="))
+            {
+                foreach (string paramset in (a[2].Split(',')))
+                {
+                    if (!String.IsNullOrEmpty(paramset))
+                    {
+                        string[] fields = paramset.Split('=');
+                        Parameters.Add(fields[0], fields[1]);

```

Comment:
shouldn't you check here that parameter name is something expected, not a random garbage string?

Reply:
technically no for expandibility. Right now the only parameter we intend to put in here is an iteration count for pbkdf, but according to the spec its supposed to be kept open so that if newer methods come along that require special additional parameters, they can be fed in.